File: scripts/wave177.py
#!/usr/bin/env python3
"""Wave 177: Add 10 new services"""
import json

JSON_PATH = 'app/data/servicesData.json'
TS_PATH = 'app/data/servicesData.ts'

NEW_SERVICES = [
  {
    "id": "ai-predictive-maintenance-iiot",
    "title": "AI Predictive Maintenance for IIoT",
    "description": "Predict industrial equipment failures before they happen using IIoT sensor data and ML. Reduce unplanned downtime and extend asset life across manufacturing, energy, and utilities.",
    "features": ["Vibration analysis", "Thermal anomaly detection", "Oil analysis prediction", "Remaining useful life estimation", "Integration with PTC ThingWorx, Siemens MindSphere", "Work order auto-generation", "Spare parts forecasting", "Maintenance scheduling optimization"],
    "benefits": ["Reduce unplanned downtime by 50%", "Extend asset life by 20%", "Lower maintenance costs", "Improve safety compliance"],
    "pricing": {"basic": "$1,999/mo", "pro": "$5,999/mo", "enterprise": "Custom"},
    "icon": "🔧", "popular": True, "category": "ai", "industry": "Manufacturing"
  },
  {
    "id": "ai-cybersecurity-threat-hunting",
    "title": "AI Cybersecurity Threat Hunting Platform",
    "description": "Proactively hunt for advanced threats using AI-driven behavioral analytics, anomaly detection, and automated investigation playbooks across endpoints, network, and cloud.",
    "features": ["Behavioral analytics (UEBA)", "Network traffic analysis", "Endpoint detection and response", "Automated threat investigation", "MITRE ATT&CK mapping", "Integration with Splunk, Sentinel, CrowdStrike", "Threat intelligence correlation", "Incident timeline reconstruction"],
    "benefits": ["Detect threats 10x faster", "Reduce dwell time to hours", "Automate Tier-1 investigation", "Meet SOC 2 and ISO 27001 requirements"],
    "pricing": {"basic": "$2,499/mo", "pro": "$7,499/mo", "enterprise": "Custom"},
    "icon": "🛡️", "popular": True, "category": "ai", "industry": "Cybersecurity"
  },
  {
    "id": "micro-saas-customer-health-score",
    "title": "HealthScore — Customer Health Scoring",
    "description": "Real-time customer health scoring for SaaS. Combine product usage, support tickets, NPS, and billing data into a single health score with automated playbooks for at-risk accounts.",
    "features": ["Composite health scoring", "Product usage tracking", "Support ticket sentiment analysis", "NPS and CSAT integration", "Automated playbooks for at-risk accounts", "Integration with Salesforce, HubSpot, Intercom", "Churn prediction alerts", "Executive health dashboards"],
    "benefits": ["Reduce churn by 25%", "Prioritize CSM efforts", "Early warning for at-risk accounts", "Data-driven retention strategy"],
    "pricing": {"basic": "$79/mo", "pro": "$249/mo", "enterprise": "$599/mo"},
    "icon": "💓", "popular": False, "category": "micro-saas", "industry": "SaaS"
  },
  {
    "id": "micro-saas-internal-wiki",
    "title": "WikiBase — Internal Knowledge Wiki",
    "description": "Modern internal wiki with AI-powered search, auto-documentation, and team collaboration. Replace Confluence with a faster, smarter knowledge base.",
    "features": ["AI-powered search and answers", "Auto-documentation from code", "Real-time collaboration", "Version history and rollback", "Integration with Slack, GitHub, Jira", "Custom templates and workflows", "Access control and permissions", "Analytics on knowledge gaps"],
    "benefits": ["Find answers 5x faster", "Reduce onboarding time", "Keep documentation alive", "Single source of truth"],
    "pricing": {"basic": "$19/mo", "pro": "$59/mo", "enterprise": "$199/mo"},
    "icon": "📚", "popular": False, "category": "micro-saas", "industry": "Productivity"
  },
  {
    "id": "it-enterprise-architecture",
    "title": "Enterprise Architecture as a Service",
    "description": "Strategic enterprise architecture consulting and tooling. Technology roadmap, application portfolio rationalization, and target architecture design for digital transformation.",
    "features": ["Application portfolio assessment", "Technology roadmap development", "Target architecture design", "Integration architecture planning", "TOGAF and ArchiMate frameworks", "Cloud readiness assessment", "Cost-benefit analysis", "Executive presentation decks"],
    "benefits": ["Align IT with business strategy", "Reduce technical debt", "Optimize technology spend", "Accelerate digital transformation"],
    "pricing": {"basic": "$4,999/mo", "pro": "$14,999/mo", "enterprise": "Custom"},
    "icon": "🏛️", "popular": False, "category": "it", "industry": "Technology"
  },
  {
    "id": "security-zero-trust-architecture",
    "title": "Zero Trust Architecture Implementation",
    "description": "End-to-end Zero Trust implementation. Identity verification, micro-segmentation, continuous monitoring, and policy enforcement across hybrid cloud and on-prem.",
    "features": ["Identity-centric security", "Micro-segmentation design", "Continuous verification", "Policy engine deployment", "Integration with Zscaler, Palo Alto, Cloudflare", "Device trust scoring", "Least-privilege access automation", "Compliance mapping (NIST 800-207)"],
    "benefits": ["Eliminate implicit trust", "Reduce attack surface by 80%", "Meet compliance requirements", "Secure hybrid workforce"],
    "pricing": {"basic": "$3,999/mo", "pro": "$11,999/mo", "enterprise": "Custom"},
    "icon": "🔒", "popular": False, "category": "security", "industry": "Technology"
  },
  {
    "id": "cloud-finops-platform",
    "title": "Cloud FinOps Platform",
    "description": "Comprehensive FinOps platform for cloud cost management. Showback/chargeback, reserved instance optimization, waste detection, and unit economics tracking.",
    "features": ["Multi-cloud cost visibility", "Showback and chargeback", "RI and Savings Plans optimization", "Waste detection and auto-remediation", "Unit economics (cost per customer, per feature)", "Integration with AWS, GCP, Azure billing", "Budget alerts and forecasting", "Executive cost dashboards"],
    "benefits": ["Reduce cloud spend by 30%", "Eliminate waste automatically", "Business-aligned cost decisions", "Accurate forecasting"],
    "pricing": {"basic": "$499/mo", "pro": "$1,499/mo", "enterprise": "Custom"},
    "icon": "💰", "popular": True, "category": "cloud", "industry": "Technology"
  },
  {
    "id": "data-realtime-streaming-platform",
    "title": "Real-Time Data Streaming Platform",
    "description": "Managed real-time data streaming with Apache Kafka, Flink, and change data capture. Ingest, process, and serve data in milliseconds for event-driven architectures.",
    "features": ["Managed Kafka clusters", "Stream processing with Flink", "Change data capture (Debezium)", "Schema registry and evolution", "Exactly-once processing guarantees", "Integration with 100+ sources and sinks", "Monitoring and alerting", "Auto-scaling"],
    "benefits": ["Process data in milliseconds", "Eliminate batch delays", "Event-driven architecture made easy", "Scale without ops burden"],
    "pricing": {"basic": "$999/mo", "pro": "$2,999/mo", "enterprise": "Custom"},
    "icon": "⚡", "popular": False, "category": "data", "industry": "Technology"
  },
  {
    "id": "ai-digital-twin-simulation",
    "title": "AI Digital Twin & Simulation Platform",
    "description": "Create and run digital twins of physical assets, processes, and entire facilities. AI-powered simulation for what-if analysis, optimization, and predictive planning.",
    "features": ["3D asset modeling", "Physics-based simulation", "AI-powered what-if analysis", "Real-time sensor integration", "Scenario comparison and optimization", "Integration with Unity, NVIDIA Omniverse", "Predictive planning dashboards", "API for custom integrations"],
    "benefits": ["Optimize operations without physical risk", "Test changes in simulation first", "Predict outcomes with AI accuracy", "Reduce planning cycles by 60%"],
    "pricing": {"basic": "$2,999/mo", "pro": "$8,999/mo", "enterprise": "Custom"},
    "icon": "🪞", "popular": False, "category": "ai", "industry": "Manufacturing"
  },
  {
    "id": "automation-intelligent-document-processing",
    "title": "Intelligent Document Processing (IDP)",
    "description": "AI-powered document processing for invoices, contracts, forms, and claims. OCR, NLP, and ML combined for touchless document workflows.",
    "features": ["Multi-format OCR (PDF, images, scans)", "NLP entity extraction", "Classification and routing", "Validation and exception handling", "Integration with SAP, Oracle, Workday", "Human-in-the-loop review", "Audit trail and compliance", "Custom model training"],
    "benefits": ["Process documents 10x faster", "Reduce manual data entry by 90%", "Improve accuracy and compliance", "Scale without adding headcount"],
    "pricing": {"basic": "$599/mo", "pro": "$1,799/mo", "enterprise": "Custom"},
    "icon": "📄", "popular": False, "category": "automation", "industry": "Technology"
  },
]

with open(JSON_PATH, 'r') as f:
    existing = json.load(f)

existing_ids = {s['id'] for s in existing}
added = 0
for svc in NEW_SERVICES:
    if svc['id'] not in existing_ids:
        entry = {
            "id": svc["id"],
            "title": svc["title"],
            "description": svc["description"],
            "features": svc["features"],
            "benefits": svc["benefits"],
            "pricing": svc["pricing"],
            "contactInfo": {
                "website": f"/services/{svc['id']}",
                "email": "kleber@ziontechgroup.com",
                "phone": "+1 302 464 0950"
            },
            "icon": svc["icon"],
            "href": f"/services/{svc['id']}",
            "popular": svc["popular"],
            "category": svc["category"],
            "industry": svc["industry"]
        }
        existing.append(entry)
        added += 1

with open(JSON_PATH, 'w') as f:
    json.dump(existing, f, indent=2, ensure_ascii=False)

print(f"JSON: Added {added} services. Total: {len(existing)}")

# Also add to TS
with open(TS_PATH, 'r') as f:
    ts = f.read()

def make_entry(svc):
    feats = json.dumps(svc['features'])
    benefits = json.dumps(svc['benefits'])
    pricing = json.dumps(svc['pricing'])
    return f"""  {{
    id: '{svc['id']}',
    title: '{svc['title']}',
    description: '{svc['description']}',
    features: {feats},
    benefits: {benefits},
    pricing: {pricing},
    contactInfo: {{website:'/services/{svc['id']}', email:'kleber@ziontechgroup.com', phone:'+1 302 464 0950'}},
    icon: '{svc['icon']}', href: '/services/{svc['id']}', popular: {str(svc['popular']).lower()}, category: '{svc['category']}', industry: '{svc['industry']}',
  }}"""

by_cat = {}
for s in NEW_SERVICES:
    cat = s['category']
    if cat not in by_cat:
        by_cat[cat] = []
    by_cat[cat].append(s)

cat_to_array = {
    'ai': 'additionalNewAiServices',
    'micro-saas': 'additionalNewMicroSaaS',
    'it': 'additionalNewItServices',
    'security': 'additionalNewSecurityServices',
    'cloud': 'additionalNewCloudServices',
    'data': 'additionalNewDataServices',
    'automation': 'additionalNewAutomationServices',
}

for cat, array_name in cat_to_array.items():
    if cat not in by_cat:
        continue
    entries = ',\n'.join(make_entry(s) for s in by_cat[cat])
    import re as re2
    pattern = rf"(export const {array_name}: Service\[\] = \[.*?)\];"
    match = re2.search(pattern, ts, re2.DOTALL)
    if match:
        old_close = match.group(0)
        existing_block = match.group(1)
        if existing_block.rstrip().endswith(']'):
            new_close = f"{existing_block}\n{entries}\n];"
        else:
            new_close = f"{existing_block},\n{entries}\n];"
        ts = ts.replace(old_close, new_close, 1)
        print(f"  Added {len(by_cat[cat])} entries to {array_name}")
    else:
        print(f"  WARNING: Could not find {array_name}")

# Fix any lone commas we introduced
lines = ts.split('\n')
new_lines = []
i = 0
while i < len(lines):
    line = lines[i]
    if line.strip() == ',' and i > 19000 and i > 0 and i < len(lines) - 1:
        prev = lines[i-1].strip()
        nxt = lines[i+1].strip()
        if prev.endswith('},') and nxt.startswith('{'):
            i += 1
            continue
    new_lines.append(line)
    i += 1

with open(TS_PATH, 'w') as f:
    f.write('\n'.join(new_lines))

print(f"TS updated with {len(NEW_SERVICES)} Wave 177 services")

File: app/data/wave177.ts
import { Service } from './serviceTypes';

export const wave177SecurityServices: Service[] = [
  {
    id: 'w177-security-incident-response',
    title: 'Incident Response Retainer',
    description: '24/7 incident response retainer for breaches, outages, and ransomware. Includes on-call IR team, forensics, containment, communication plan, and post-incident reporting.',
    features: ['24/7 on-call IR team', 'Forensic evidence collection', 'Containment and eradication', 'Stakeholder communication plan', 'Post-incident report'],
    benefits: ['Reduce breach dwell time', 'Meet regulatory disclosure timelines', 'Preserve evidence for insurance and legal', 'Restore trust with stakeholders'],
    pricing: { basic: '$1,499/mo', pro: '$3,999/mo', enterprise: 'Custom' },
    contactInfo: { website: '/services/w177-security-incident-response', email: 'kleber@ziontechgroup.com', phone: '+1 302 464 0950' },
    icon: '🚨', href: '/services/w177-security-incident-response', popular: false, category: 'security', industry: 'Security Services',
  },
];

export const wave177CloudServices: Service[] = [
  {
    id: 'w177-cloud-cost-optimization',
    title: 'Cloud Cost Optimization',
    description: 'Reduce cloud spend 20-40% with rightsizing, reserved instance planning, and waste detection across AWS, Azure, and GCP.',
    features: ['Cost visibility and dashboards', 'Rightsizing recommendations', 'Reserved instance planning', 'Waste detection and cleanup', 'Multi-cloud normalization', 'Anomaly alerts and budgets'],
    benefits: ['Lower spend without performance loss', 'Visibility into every resource', 'Protect against billing surprises', 'Finance and engineering alignment'],
    pricing: { basic: '$499/mo', pro: '$1,499/mo', enterprise: 'Custom' },
    contactInfo: { website: '/services/w177-cloud-cost-optimization', email: 'kleber@ziontechgroup.com', phone: '+1 302 464 0950' },
    icon: '💸', href: '/services/w177-cloud-cost-optimization', popular: false, category: 'cloud', industry: 'Cloud',
  },
];

export const wave177DataServices: Service[] = [
  {
    id: 'w177-data-realtime-pipeline',
    title: 'Realtime Data Pipeline',
    description: 'Kafka/Flink-based streaming pipelines for operational analytics, CDC, and event-driven architectures with schema governance.',
    features: ['Change data capture (CDC)', 'Stream processing and enrichment', 'Schema registry and governance', 'Dead-letter handling and replay', 'Operational dashboards', 'Data quality checks'],
    benefits: ['Insights within seconds, not hours', 'Reduce batch-window complexity', 'Governed data as it flows', 'Resilient pipelines with replay'],
    pricing: { basic: '$1,199/mo', pro: '$3,499/mo', enterprise: 'Custom' },
    contactInfo: { website: '/services/w177-data-realtime-pipeline', email: 'kleber@ziontechgroup.com', phone: '+1 302 464 0950' },
    icon: '🔄', href: '/services/w177-data-realtime-pipeline', popular: false, category: 'data', industry: 'Data Engineering',
  },
];

export const wave177AutomationServices: Service[] = [
  {
    id: 'w177-automation-process-mining',
    title: 'Process Mining & Automation Discovery',
    description: 'Discover real workflows from system logs, model optimization paths, and generate automation candidates with ROI estimates.',
    features: ['Process discovery from logs', 'Bottleneck identification', 'Automation opportunity scoring', 'ROI estimation', 'Flow generation for RPA/iPaaS', 'Change impact analysis'],
    benefits: ['Automate what matters most first', 'Prove ROI before building', 'Reduce manual assessment work', 'Continuous improvement loop'],
    pricing: { basic: '$799/mo', pro: '$2,199/mo', enterprise: 'Custom' },
    contactInfo: { website: '/services/w177-automation-process-mining', email: 'kleber@ziontechgroup.com', phone: '+1 302 464 0950' },
    icon: '⛏️', href: '/services/w177-automation-process-mining', popular: false, category: 'automation', industry: 'Automation',
  },
];

export const wave177MicroSaasServices: Service[] = [
  {
    id: 'w177-micro-saas-workos-auth',
    title: 'WorkOS Auth Kit',
    description: 'Drop-in SSO, directory sync, and audit logs for micro-SaaS: SAML, OIDC, SCIM, and admin APIs.',
    features: ['SSO with SAML/OIDC', 'SCIM directory sync', 'Audit logs and events', 'Admin dashboard', 'Org and role management', 'Security hardening support'],
    benefits: ['Sell to enterprise buyers faster', 'Reduce identity engineering cost', 'Pass SOC2/ISO audits easier', 'Standard auth with less code'],
    pricing: { basic: '$39/mo', pro: '$99/mo', enterprise: '$299/mo' },
    contactInfo: { website: '/services/w177-micro-saas-workos-auth', email: 'kleber@ziontechgroup.com', phone: '+1 302 464 0950' },
    icon: '🧩', href: '/services/w177-micro-saas-workos-auth', popular: true, category: 'micro-saas', industry: 'Developer Tools',
  },
];

export const wave177ItServices: Service[] = [
  {
    id: 'w177-it-backup-recovery',
    title: 'Backup & Disaster Recovery',
    description: 'Policy-driven backups, point-in-time recovery, and DR runbooks for SaaS and infrastructure with RPO/RTO guarantees.',
    features: ['Policy-driven backup schedules', 'Point-in-time recovery', 'DR runbooks and testing', 'Cross-region replication', 'RPO/RTO reporting', 'Compliance retention'],
    benefits: ['Reduce data loss risk', 'Meet recovery objectives', 'Operationalize DR with runbooks', 'Retention compliance without guesswork'],
    pricing: { basic: '$399/mo', pro: '$1,199/mo', enterprise: 'Custom' },
    contactInfo: { website: '/services/w177-it-backup-recovery', email: 'kleber@ziontechgroup.com', phone: '+1 302 464 0950' },
    icon: '🗄️', href: '/services/w177-it-backup-recovery', popular: false, category: 'it', industry: 'Infrastructure',
  },
];

export const wave177AiServices: Service[] = [
  {
    id: 'w177-ai-meeting-intelligence',
    title: 'Meeting Intelligence Platform',
    description: 'Record, transcribe, summarize meetings, extract action items, and push them to CRM/project tools automatically.',
    features: ['Recording and transcription', 'Summaries and highlights', 'Action item extraction', 'Sentiment and engagement signals', 'CRM/project tool sync', 'Meeting search and replay'],
    benefits: ['Never miss a commitment', 'Summarize meetings in minutes', 'Automate follow-ups', 'Improve team alignment'],
    pricing: { basic: '$49/mo', pro: '$149/mo', enterprise: '$499/mo' },
    contactInfo: { website: '/services/w177-ai-meeting-intelligence', email: 'kleber@ziontechgroup.com', phone: '+1 302 464 0950' },
    icon: '🎯', href: '/services/w177-ai-meeting-intelligence', popular: true, category: 'ai', industry: 'Productivity',
  },
];
